components/turret.py

```diff
@@ -8,34 +8,38 @@
 class Index(Enum):
     NOT_FOUND = 0
     CENTRE = 1
-    # RIGHT = 2
-    # LEFT = 3
-    # BACK = 4
+    RIGHT = 2
+    LEFT = 3
+    BACK = 4
 
 
 class Turret:
-    # TODO - There will be several (4 to 6) hall-effect sensors distributed
-    # around the turrent ring, with one of them aligned with an azimuth of 0
-    # relative to the robot. Right now there is only one hall-effect sensor at
-    # that point.
+    # TODO - There should be 4 indexes total: left, right, front and rear hall-effect
+    # sensors. Right now there is only one hall-effect sensor at the front.
     # left_index: wpilib.DigitalInput
     # right_index: wpilib.DigitalInput
     centre_index: wpilib.DigitalInput
-    HALL_EFFECT_CLOSED = False
-
     motor: ctre.WPI_TalonSRX
 
     # Possible states
-    IDLE = 0
-    SLEWING = 1
-    FINDING_INDEX = 2
+    SLEWING = 0
+    SCANNING = 1
 
     # Constants for Talon on the turret
     COUNTS_PER_MOTOR_REV = 4096
     GEAR_REDUCTION = 160 / 18
     COUNTS_PER_TURRET_REV = COUNTS_PER_MOTOR_REV * GEAR_REDUCTION
     COUNTS_PER_TURRET_RADIAN = COUNTS_PER_TURRET_REV / math.tau
 
+    INDEX_POSITIONS = {
+        Index.CENTRE: 0,
+        Index.LEFT: int(math.pi / 2 * COUNTS_PER_TURRET_RADIAN),
+        Index.RIGHT: int(-math.pi / 2 * COUNTS_PER_TURRET_RADIAN),
+        Index.BACK: int(math.pi * COUNTS_PER_TURRET_RADIAN),
+    }
```

Comment:
Possibly crazy idea: why not set the enum values to these?

Reply:
We could. I wonder if that might be confusing - they get returned to signify which sensor is tripped, but also get treated as the number of counts to set.
I'm indifferent (but not inclined to change it at this point).

Reply:
No, I don't think they should mean two things. Overloading is often a source of great pain.

I'm a bit more concerned that these are prescriptive rather than descriptive. They rely on the placement of the sensors being precise to within a single encoder count, which is unlikely. I think we should measure where they are exactly on the turret once they are there, in encoder counts, and use those numbers.